chainer/initializers/constant.py

```diff
@@ -30,17 +30,23 @@ def __call__(self, array):
         if len(shape) != 2 or shape[0] != shape[1]:
             raise ValueError('Identity matrix initialization can only be used '
                              'for 2D squared matrices.')
-        array[...] = 0
-        xp = backend.get_array_module(array)
-        xp.fill_diagonal(array, self.scale)
+
+        device = backend.get_device_from_array(array)
+        # ideep64 does not support fill_diagonal
+        if isinstance(device, chainer.backends.intel64.Intel64Device):
+            array = device.xp.array(
```

Comment:
`array` should be updated in place.

Reply:
Ok! I copied the numpy code for fill_diagonal and use flat() instead of flat as it was where the problem happened with ideep64